File: client/src/boundary/interactor/ShipD2BConverter.test.ts
import Ship from "domain/Ship";
import BoundaryShip from "boundary/model/BoundaryShip";
import { ShipD2BConverter } from "./ShipD2BConverter";

describe(ShipD2BConverter, () => {
  let converter: ShipD2BConverter;

  beforeEach(() => {
    converter = new ShipD2BConverter();
  });

  it("Converts Ship model to BoundaryShip model", () => {
    const inputShip: Ship = new Ship(0, [{ x: 1, y: 1 }], 25);
    const expectedShip: BoundaryShip = new BoundaryShip(
      0,
      [{ x: 1, y: 1 }],
      25,
      false
    );

    const convertedShip: BoundaryShip = converter.convert(inputShip);

    expect(convertedShip.hits).toEqual(expectedShip.hits);
    expect(convertedShip.length).toEqual(expectedShip.length);
    expect(convertedShip.destroyed).toEqual(expectedShip.destroyed);
    expect(convertedShip.coordinates).toEqual(expectedShip.coordinates);
  });
});

File: server/battleships/use_cases/implementation/src/test/java/eu/optas/use_cases/implementation/ShipD2BConverterTest.java
package eu.optas.use_cases.implementation;

import eu.optas.domain.Ship;
import eu.optas.use_cases.api.BoundaryShip;
import eu.optas.utils.Coordinates;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class ShipD2BConverterTest {

    @Test
    void convert() {
        ShipD2BConverter shipD2BConverter = new ShipD2BConverter();
        Coordinates coordinates = new Coordinates(1, 1);
        List<Coordinates> coordinatesList = new ArrayList<>();
        coordinatesList.add(coordinates);

        Ship inputShip = new Ship(5, coordinatesList, 0, false);

        BoundaryShip convertedShip = shipD2BConverter.convert(inputShip);

        assertThat(convertedShip).usingRecursiveComparison().isEqualTo(inputShip);
    }
}